make-util.js

```diff
@@ -16,7 +16,7 @@ var downloadPath = path.join(__dirname, '_download');
 // list of .NET culture names
 var cultureNames = ['cs', 'de', 'es', 'fr', 'it', 'ja', 'ko', 'pl', 'pt-BR', 'ru', 'tr', 'zh-Hans', 'zh-Hant'];
 
-var allowedTypescriptVersions = ['4.0.2', '4.0.8'];
+var allowedTypescriptVersions = ['4.0.2', '4.0.8','5.1.6'];
```

Comment:
>allowedTypescriptVersions [](http://example.com/codeflow?start=4&length=25)

we should ensure that 4.0.2/8 is used when building 'Default' tasks and 5.1.6 when building 'Node20' tasks

we can also drop 8 but I'm okay with saving that for another PR

Reply:
This allowedTypescriptVersions is tsc versions we allowed on base tasks. We're able to build Node 20 tasks using tsc 4.0.8 after running nvm use 20 to switch node version.

Reply:
My point is to keep things consistent, we should only allow Node20 tasks to be built using 5.1.6 and sub-Node20 tasks to be built using 4.0.8 or 4.0.2.   No mix and matching.  

Maybe we can save for a follow up PR to avoid too many iterations on this one.

Reply:
We don't need to have tsc 5.1.6 to build node20 tasks. There're 2 tsc version we're talking, one is the tsc version on the system (version showed up in package.json on root folder, 4.0.2/4.0.8), the other is tsc on base tasks(we use 5.1.6 for node20). We don't need to change tsc version on the system to build node20 tasks. The only change would be node engine version.